wagtail/admin/views/generic/models.py

```diff
@@ -266,6 +276,10 @@ def columns(self):
                 column = self._get_title_column(field)
             else:
                 column = self._get_custom_column(field)
+
+            if isinstance(column, LocaleColumn) and not self.i18n_enabled:
```

Comment:
Not super happy with this, but we can't initialise `list_display` on init nor make it a (cached) property, because `ModelViewSet` / `SnippetViewSet` fetch `list_display` directly of the index view class, rather than its instance. And `i18n_enabled` checks both the setting and the fact that the given model is translatable